src/mare/compiler/jumps.cr

```diff
@@ -11,87 +11,232 @@
 # This pass produces no output state.
 #
 module Mare::Compiler::Jumps
-  FLAG_ALWAYS_ERROR = 0x01_u8
-  FLAG_MAYBE_ERROR  = 0x02_u8
+  FLAG_ALWAYS_ERROR = 0b1_u8
+  FLAG_MAYBE_ERROR  = 0b10_u8
+  FLAG_ALWAYS_RETURN = 0b100_u8
+  FLAG_MAYBE_RETURN  = 0b1000_u8
+  FLAG_ALWAYS_BREAK = 0b10000_u8
+  FLAG_MAYBE_BREAK  = 0b100000_u8
+  FLAG_ALWAYS_CONTINUE = 0b1000000_u8
+  FLAG_MAYBE_CONTINUE  = 0b10000000_u8
+
+  alias JumpKind = AST::Jump::Kind
 
   struct Analysis
+    getter catches
+
     def initialize
-      @flags = {} of AST::Node => UInt8
+      @flags = {} of AST::Node => {UInt8, Bool}
```

Comment:
Seems simpler to just make this a `UInt16`, rather than a tuple of `{UInt8, Bool}` - it doesn't save any space, since the alignment of the data will force it to the next wider memory width anyway.

Reply:
I'll change it